statsmodels/duration/hazard_regression.py

```diff
@@ -402,10 +402,19 @@ def from_formula(cls, formula, data, status=None, entry=None,
         if isinstance(offset, str):
             offset = data[offset]
 
+        import re
+        terms = re.split("[+\-~]", formula)
+        for term in terms:
+            term = term.strip()
+            if term in ("0", "1"):
+                import warnings
+                warnings.warn("PHReg formulas should not include any '0' or '1' terms")
```

Comment:
I'm not sure it's easy, worth it to improve.
if there are no categoricals, than `0+` or `-1` would be the right thing to do to avoid the drop columns after creating them.